extensions/amp-video/0.1/test-e2e/test-amp-video-flexible-bitrate.js

```diff
@@ -0,0 +1,232 @@
+  UNLOAD: 'u',
+};
+
+describes.endtoend(
+  'amp-video flexible bitrate',
+  {
+    fixture: 'amp-video/videos-cdn.html',
+    environments: ['single'],
+    experiments: ['flexible-bitrate'],
+    initialRect: {width: VIEWPORT.WIDTH, height: VIEWPORT.HEIGHT},
+  },
+  (env) => {
+    let controller;
+    let story;
+    let debugField;
+
+    beforeEach(async () => {
+      controller = env.controller;
+
+      story = await controller.findElement('amp-story.i-amphtml-story-loaded');
+      await controller.findElement('amp-story-page#page-1[active]');
+
+      debugField = await controller.findElement('input#flexible-bitrate-debug');
+    });
+
+    /**
+     * Tests for first load
+     */
+
+    it('when loading the page with good connection, the best quality source is played', async () => {
+      const video1El = await controller.findElement('#video1 video');
+      await expect(
+        await controller.getElementProperty(video1El, 'currentSrc')
+      ).contains('#high');
+    });
+
+    it("when on the first page, a further video doesn't load", async () => {
+      const video4El = await controller.findElement('#video4 video');
+      await expect(
+        await controller.getElementProperty(video4El, 'currentSrc')
+      ).to.equal('');
+    });
+
+    /**
+     * Tests for downgrades on active video
+     */
+
+    it('when a video buffers, the video changes to a lower quality source', async () => {
+      await forceEventOnVideo(VIDEO_EVENTS.UNLOAD, 1);
+
+      const video1El = await controller.findElement('#video1 video');
+
+      await forceEventOnVideo(VIDEO_EVENTS.DOWNGRADE, 1);
+
+      await expect(
+        await controller.getElementProperty(video1El, 'currentSrc')
+      ).contains('#med');
+    });
+
+    it('when a video buffers, the low quality source starts playing at the same place the previous source was stopped at', async () => {
+      const video1El = await controller.findElement('#video1 video');
+
+      await sleep(100);
+
+      const currentTime = await controller.getElementProperty(
+        video1El,
+        'currentTime'
+      );
+
+      await forceEventOnVideo(VIDEO_EVENTS.UNLOAD, 1);
+      await forceEventOnVideo(VIDEO_EVENTS.DOWNGRADE, 1);
+
+      await expect(
+        await controller.getElementProperty(video1El, 'currentTime')
+      ).to.be.gte(currentTime);
+    });
+
+    it("when a video buffers many times, it doesn't keep reloading", async () => {
+      await forceEventOnVideo(VIDEO_EVENTS.UNLOAD, 1);
+      await forceEventOnVideo(VIDEO_EVENTS.DOWNGRADE, 1);
+      await forceEventOnVideo(VIDEO_EVENTS.DOWNGRADE, 1);
+      await forceEventOnVideo(VIDEO_EVENTS.DOWNGRADE, 1);
+
+      const video1El = await controller.findElement(
+        '#video1 video.i-amphtml-pool-video'
+      );
+
+      await expect(
+        await controller.getElementProperty(video1El, 'currentSrc')
+      ).contains('#low');
+      await expect(await controller.getElementProperty(video1El, 'paused')).is
```

Comment:
How does this test that it didn't reload? Even if calling `load()` non stop `paused` would be false